src/grafana_agent.py

```diff
@@ -868,6 +868,88 @@ def _receiver_config(protocol: str):
 
         return config
 
+    @property
+    def _tracing_sampling(self) -> Dict[str, Any]:
+        return {
+            "policies": [
+                {
+                    "name": "error-traces-policy",
+                    "type": "and",
+                    "and": {
+                        "and_sub_policy": [
+                            {
+                                "name": "trace-status-policy",
+                                "type": "status_code",
+                                "status_code": {"status_codes": ["ERROR"]},
+                            },
+                            {
+                                "name": "probabilistic-policy",
+                                "type": "probabilistic",
+                                "probabilistic": {
+                                    "sampling_percentage": self.config.get(
```

Comment:
blindly putting the config value in here feels icky. What if the user set `9123213123123129837`? what if the user set `-24` (is that even juju-possible)?
I think we need a little bit of validation and parsing.
I'd propose adding a `self._sampling_percentages` preprocessor using pydantic to cast all values in the 0,100 range (and perhaps even round to 3/4 decimals?) for cleanliness.

Reply:
As discussed separately, grafana-agent does normalise the values but I've added a remark in the configuration properties description so that it's clear.